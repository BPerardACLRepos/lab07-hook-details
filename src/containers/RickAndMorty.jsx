import React from 'react';

import CharacterList from '../components/characters/CharacterList';
import Loading from '../components/characters/Loading';
import Pagination from '../components/characters/Pagination';

import { handleList } from '../hooks/pageAndCharacters';

const RickAndMorty = () => {
    const { loading, page, lastPage, characters, setPage } = handleList();

    if (loading) return <Loading />;

    return (
        <>
            <Pagination
                page={page}
                lastPage={lastPage}
                setPage={setPage}
            />
            <CharacterList characters={characters} />
        </>
    );
};

export default RickAndMorty;