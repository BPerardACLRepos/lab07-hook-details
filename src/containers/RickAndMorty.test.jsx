import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import rickAndMortyCharacters from '../../fixtures/rickAndMortyCharacters.json';
import RickAndMorty from './RickAndMorty';

const server = setupServer(
    rest.get(`https://rickandmortyapi.com/api/character?page=1`,
        (req, res, ctx) => {
            return res(
                ctx.json(rickAndMortyCharacters)
            );
        })
);

jest.useFakeTimers();

describe('RickAndMorty Container custom hooks', () => {
    beforeAll(() => server.listen());
    afterAll(() => server.close());

    it('Show loading, then fire button to new page, and render return as list items', async () => {
        render(<RickAndMorty />);

        screen.getByText('Loading...');


        const button = await screen.findByRole('button', { name: 'next-button' });
        fireEvent.click(button);

        screen.getByText('Loading...');

        const ul = await screen.findByRole('list', { name: 'character-list' });
        expect(ul).toMatchSnapshot();

    });
});
