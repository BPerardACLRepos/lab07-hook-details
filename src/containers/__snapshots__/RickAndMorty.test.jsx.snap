// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RickAndMorty Container custom hooks Show loading, then fire button to new page, and render return as list items 1`] = `
<ul
  aria-label="character-list"
>
  <li>
    <a
      href="/21"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Aqua Morty
           (
          21
          )
        </h3>
        <figure>
          <img
            alt="Aqua Morty"
            src="https://rickandmortyapi.com/api/character/avatar/21.jpeg"
          />
          <figcaption>
            <p>
              unknown
               - 
              Humanoid
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/22"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Aqua Rick
           (
          22
          )
        </h3>
        <figure>
          <img
            alt="Aqua Rick"
            src="https://rickandmortyapi.com/api/character/avatar/22.jpeg"
          />
          <figcaption>
            <p>
              unknown
               - 
              Humanoid
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/23"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Arcade Alien
           (
          23
          )
        </h3>
        <figure>
          <img
            alt="Arcade Alien"
            src="https://rickandmortyapi.com/api/character/avatar/23.jpeg"
          />
          <figcaption>
            <p>
              unknown
               - 
              Alien
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/24"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Armagheadon
           (
          24
          )
        </h3>
        <figure>
          <img
            alt="Armagheadon"
            src="https://rickandmortyapi.com/api/character/avatar/24.jpeg"
          />
          <figcaption>
            <p>
              Alive
               - 
              Alien
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/25"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Armothy
           (
          25
          )
        </h3>
        <figure>
          <img
            alt="Armothy"
            src="https://rickandmortyapi.com/api/character/avatar/25.jpeg"
          />
          <figcaption>
            <p>
              Dead
               - 
              unknown
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/26"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Arthricia
           (
          26
          )
        </h3>
        <figure>
          <img
            alt="Arthricia"
            src="https://rickandmortyapi.com/api/character/avatar/26.jpeg"
          />
          <figcaption>
            <p>
              Alive
               - 
              Alien
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/27"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Artist Morty
           (
          27
          )
        </h3>
        <figure>
          <img
            alt="Artist Morty"
            src="https://rickandmortyapi.com/api/character/avatar/27.jpeg"
          />
          <figcaption>
            <p>
              Alive
               - 
              Human
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/28"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Attila Starwar
           (
          28
          )
        </h3>
        <figure>
          <img
            alt="Attila Starwar"
            src="https://rickandmortyapi.com/api/character/avatar/28.jpeg"
          />
          <figcaption>
            <p>
              Alive
               - 
              Human
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/29"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Baby Legs
           (
          29
          )
        </h3>
        <figure>
          <img
            alt="Baby Legs"
            src="https://rickandmortyapi.com/api/character/avatar/29.jpeg"
          />
          <figcaption>
            <p>
              Alive
               - 
              Human
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/30"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Baby Poopybutthole
           (
          30
          )
        </h3>
        <figure>
          <img
            alt="Baby Poopybutthole"
            src="https://rickandmortyapi.com/api/character/avatar/30.jpeg"
          />
          <figcaption>
            <p>
              Alive
               - 
              Poopybutthole
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/31"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Baby Wizard
           (
          31
          )
        </h3>
        <figure>
          <img
            alt="Baby Wizard"
            src="https://rickandmortyapi.com/api/character/avatar/31.jpeg"
          />
          <figcaption>
            <p>
              Dead
               - 
              Alien
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/32"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Bearded Lady
           (
          32
          )
        </h3>
        <figure>
          <img
            alt="Bearded Lady"
            src="https://rickandmortyapi.com/api/character/avatar/32.jpeg"
          />
          <figcaption>
            <p>
              Dead
               - 
              Alien
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/33"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Beebo
           (
          33
          )
        </h3>
        <figure>
          <img
            alt="Beebo"
            src="https://rickandmortyapi.com/api/character/avatar/33.jpeg"
          />
          <figcaption>
            <p>
              Dead
               - 
              Alien
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/34"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Benjamin
           (
          34
          )
        </h3>
        <figure>
          <img
            alt="Benjamin"
            src="https://rickandmortyapi.com/api/character/avatar/34.jpeg"
          />
          <figcaption>
            <p>
              Alive
               - 
              Poopybutthole
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/35"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Bepisian
           (
          35
          )
        </h3>
        <figure>
          <img
            alt="Bepisian"
            src="https://rickandmortyapi.com/api/character/avatar/35.jpeg"
          />
          <figcaption>
            <p>
              Alive
               - 
              Alien
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/36"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Beta-Seven
           (
          36
          )
        </h3>
        <figure>
          <img
            alt="Beta-Seven"
            src="https://rickandmortyapi.com/api/character/avatar/36.jpeg"
          />
          <figcaption>
            <p>
              Alive
               - 
              Alien
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/37"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Beth Sanchez
           (
          37
          )
        </h3>
        <figure>
          <img
            alt="Beth Sanchez"
            src="https://rickandmortyapi.com/api/character/avatar/37.jpeg"
          />
          <figcaption>
            <p>
              Alive
               - 
              Human
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/38"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Beth Smith
           (
          38
          )
        </h3>
        <figure>
          <img
            alt="Beth Smith"
            src="https://rickandmortyapi.com/api/character/avatar/38.jpeg"
          />
          <figcaption>
            <p>
              Alive
               - 
              Human
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/39"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Beth Smith
           (
          39
          )
        </h3>
        <figure>
          <img
            alt="Beth Smith"
            src="https://rickandmortyapi.com/api/character/avatar/39.jpeg"
          />
          <figcaption>
            <p>
              Alive
               - 
              Human
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/40"
    >
      <div
        aria-label="character-div"
      >
        <h3>
          Beth's Mytholog
           (
          40
          )
        </h3>
        <figure>
          <img
            alt="Beth's Mytholog"
            src="https://rickandmortyapi.com/api/character/avatar/40.jpeg"
          />
          <figcaption>
            <p>
              Dead
               - 
              Mythological Creature
            </p>
          </figcaption>
        </figure>
      </div>
    </a>
  </li>
</ul>
`;
